components/ILIAS/Chatroom/classes/class.BannedUsersTable.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Data;
+use ILIAS\UI;
+use Psr\Http\Message\ServerRequestInterface;
+use ILIAS\HTTP\Services;
+
+class BannedUsersTable implements UI\Component\Table\DataRetrieval
+{
+    protected ServerRequestInterface $request;
+    protected Data\Factory $data_factory;
+    private ?array $records = null;
```

Comment:
Can you try to specify the shape of the array here?

`list<array<string, mixed>>|null`